### ISSUE TITLE
Follow-up on does XGBOOST do random sampling when subsample = 1

### ISSUE THREAD

**Issue by sjain777** [dmlc/xgboost#3260]
Hello, as a follow-up on issue no. 266 which answers that NO random sampling on the number of training data rows is performed when subsampel = 1, my question is what then results in different trees in each iteration of  nrounds ( set to greater than 1) when both subsample = 1 (default) and colsample_bytree = 1 (default). Thanks!

**Comment by hcho3** [comment]
Even without random sampling, you'd get different tree in each iteration because of how boosting works. In gradient boosting, different decision trees are trained so that they are complementary to each other.
